src/bech32.cpp

```diff
@@ -161,18 +161,20 @@ std::pair<std::string, data> Decode(const std::string& str) {
     for (size_t i = 0; i < str.size(); ++i) {
         unsigned char c = str[i];
         if (c < 33 || c > 126) return {};
-        if (c >= 'a' && c <= 'z') lower = true;
-        if (c >= 'A' && c <= 'Z') upper = true;
+        // Mixing upper and lowercase is not OK.
+        if (!lower && c >= 'a' && c <= 'z') { if (upper) return {}; lower = true; }
```

Comment:
Please follow the coding style: any `if` that has anything but just a single-statement then body must use braces and indentation.